art/estimators/estimator.py

```diff
@@ -492,11 +492,30 @@ def get_activations(self, x, layer, batch_size):
         :type layer: `int` or `str`
         :param batch_size: Batch size.
         :type batch_size: `int`
+        :param intermediate: if true, return the intermediate tensor representation of the activation
+        :type intermediate: `bool`
         :return: The output of `layer`, where the first dimension is the batch size corresponding to `x`.
         :rtype: `np.ndarray`
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def custom_gradient(self, nn_function, tensors, input_values):
+        """
+        Returns the gradient of the nn_function with respect to vars
+
+        :param nn_function: an intermediate tensor representation of the gradient function
+        :param vars: the variables to differentiate
+        :type vars: `list`
+        :return: the gradient of the function w.r.t vars
+        :rtype: `np.ndarray`
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_input_layer(self):
+        raise NotImplementedError
+
```

Comment:
Please remove these lines because `custom_gradient` is so far only part of `KerasClassifier` and not the Estimator API.